### DISCUSSION TITLE
82599-sfp can not up

### DISCUSSION THREAD

**Discussion by nanotouch** [ipxe/ipxe#929] [General]
can not up，how to deal，someone can help me，legacy mode
![img_v2_ace9fb69-142c-4801-a2da-95358a250b8g](https://user-images.githubusercontent.com/52226706/228115371-b5a9ed43-ceb6-48ef-a068-6683f3ded170.jpg)

**Comment by NiKiZe** [comment]
Moved to discussions until we get data to know what to fix, if anything.

https://ipxe.org/38086101

When you do dhcp, let it fail, followed by ifstat, what does it show?
How many ports does the NIC have?

Does the link work unchanged in other OS?
Do you get any link up?
Which SFP module are you using?

**Reply by nanotouch** [reply]
NIC have 2 ports
using optical-to-electrical module，I don't know if it is caused by the incompatibility of the optical-to-electrical module. I have tried to replace other brands of optical-to-electrical modules, and the problem is the same.
![img_v2_5c60ea79-33ea-4ef3-8f68-0d2c88473a4g](https://user-images.githubusercontent.com/52226706/228132057-5aca8bf6-748a-456b-98af-46a8d76e6e87.jpg)

**Reply by stappersg** [reply]
https://ipxe.org/38086101

**Reply by NiKiZe** [reply]
Please answer all the questions

Again:
Does it work in any OS? (Asking to verify that the link works in some manner)
Which link speed do you have? (100m might get more issues than 1gig)
Which SFP module are you using? I'm especially looking for which link speed it supports, you will have to look at the module which generally have a type written on it.

Have you tried undionly.kpxe?

**Reply by nanotouch** [reply]
i use undionly.pxe ,solve,tanks

**Reply by stappersg** [reply]
I think that **if** switching to `undionly.pxe` was the **only** change, that the original issue of _iPXE_ **not** seeing network link still exists. If so, it would be a good thing to make a fresh start.

**Reply by alarig** [reply]
I’m running into the same issue with a 82599en-sfp NIC. I’ve build an ISO by hand using `undionly.kpxe` instead of `ipxe.lkrn` and it boots.

Here is what lspci looks like from a livecd:
```
livecd ~ # lspci -k
00:00.0 Host bridge: Intel Corporation Xeon E3-1200 v3 Processor DRAM Controller (rev 06)
	Subsystem: Super Micro Computer Inc Xeon E3-1200 v3 Processor DRAM Controller
00:01.0 PCI bridge: Intel Corporation Xeon E3-1200 v3/4th Gen Core Processor PCI Express x16 Controller (rev 06)
	Subsystem: Super Micro Computer Inc Xeon E3-1200 v3/4th Gen Core Processor PCI Express x16 Controller
	Kernel driver in use: pcieport
00:14.0 USB controller: Intel Corporation 8 Series/C220 Series Chipset Family USB xHCI (rev 05)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family USB xHCI
	Kernel driver in use: xhci_hcd
	Kernel modules: xhci_pci
00:16.0 Communication controller: Intel Corporation 8 Series/C220 Series Chipset Family MEI Controller #1 (rev 04)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family MEI Controller
00:16.1 Communication controller: Intel Corporation 8 Series/C220 Series Chipset Family MEI Controller #2 (rev 04)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family MEI Controller
00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 05)
	DeviceName:  Intel Ethernet i217LM #2
	Subsystem: Super Micro Computer Inc Ethernet Connection I217-LM
	Kernel driver in use: e1000e
	Kernel modules: e1000e
00:1a.0 USB controller: Intel Corporation 8 Series/C220 Series Chipset Family USB EHCI #2 (rev 05)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family USB EHCI
	Kernel driver in use: ehci-pci
	Kernel modules: ehci_pci
00:1c.0 PCI bridge: Intel Corporation 8 Series/C220 Series Chipset Family PCI Express Root Port #1 (rev d5)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family PCI Express Root Port
	Kernel driver in use: pcieport
00:1c.1 PCI bridge: Intel Corporation 8 Series/C220 Series Chipset Family PCI Express Root Port #2 (rev d5)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family PCI Express Root Port
	Kernel driver in use: pcieport
00:1d.0 USB controller: Intel Corporation 8 Series/C220 Series Chipset Family USB EHCI #1 (rev 05)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family USB EHCI
	Kernel driver in use: ehci-pci
	Kernel modules: ehci_pci
00:1f.0 ISA bridge: Intel Corporation C224 Series Chipset Family Server Standard SKU LPC Controller (rev 05)
	Subsystem: Super Micro Computer Inc C224 Series Chipset Family Server Standard SKU LPC Controller
	Kernel driver in use: lpc_ich
	Kernel modules: lpc_ich
00:1f.2 SATA controller: Intel Corporation 8 Series/C220 Series Chipset Family 6-port SATA Controller 1 [AHCI mode] (rev 05)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family 6-port SATA Controller 1 [AHCI mode]
	Kernel driver in use: ahci
	Kernel modules: ahci
00:1f.3 SMBus: Intel Corporation 8 Series/C220 Series Chipset Family SMBus Controller (rev 05)
	Subsystem: Super Micro Computer Inc 8 Series/C220 Series Chipset Family SMBus Controller
	Kernel driver in use: i801_smbus
	Kernel modules: i2c_i801
00:1f.6 Signal processing controller: Intel Corporation 8 Series Chipset Family Thermal Management Controller (rev 05)
	Subsystem: Super Micro Computer Inc 8 Series Chipset Family Thermal Management Controller
01:00.0 Ethernet controller: Intel Corporation 82599 10 Gigabit Network Connection (rev 01)
	Subsystem: Super Micro Computer Inc 82599 10 Gigabit Network Connection
	Kernel driver in use: ixgbe
	Kernel modules: ixgbe
02:00.0 PCI bridge: ASPEED Technology, Inc. AST1150 PCI-to-PCI Bridge (rev 03)
	Subsystem: Super Micro Computer Inc AST1150 PCI-to-PCI Bridge
03:00.0 VGA compatible controller: ASPEED Technology, Inc. ASPEED Graphics Family (rev 30)
	DeviceName:  ASPEED Video AST2400
	Subsystem: Super Micro Computer Inc ASPEED Graphics Family
04:00.0 Ethernet controller: Intel Corporation I210 Gigabit Network Connection (rev 03)
	DeviceName:  Intel Ethernet i210AT #1
	Subsystem: Super Micro Computer Inc I210 Gigabit Network Connection
	Kernel driver in use: igb
	Kernel modules: igb
livecd ~ # ip l
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eno1: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN mode DEFAULT group default qlen 1000
    link/ether 0c:c4:7a:7d:9f:a0 brd ff:ff:ff:ff:ff:ff
    altname enp4s0
3: eno2: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc pfifo_fast state DOWN mode DEFAULT group default qlen 1000
    link/ether 0c:c4:7a:7d:9f:a1 brd ff:ff:ff:ff:ff:ff
    altname enp0s25
4: enp1s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 0c:c4:7a:bb:17:b2 brd ff:ff:ff:ff:ff:ff
livecd ~ # 
```

**Comment by stappersg** [comment]
On Wed, May 10, 2023 at 12:33:21PM -0700, Alarig Le Lay wrote:
> 00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 05)
> 	DeviceName:  Intel Ethernet i217LM #2
> 	Subsystem: Super Micro Computer Inc Ethernet Connection I217-LM
> 	Kernel driver in use: e1000e
> 	Kernel modules: e1000e
> 01:00.0 Ethernet controller: Intel Corporation 82599 10 Gigabit Network Connection (rev 01)
> 	Subsystem: Super Micro Computer Inc 82599 10 Gigabit Network Connection
> 	Kernel driver in use: ixgbe
> 	Kernel modules: ixgbe
> 04:00.0 Ethernet controller: Intel Corporation I210 Gigabit Network Connection (rev 03)
> 	DeviceName:  Intel Ethernet i210AT #1
> 	Subsystem: Super Micro Computer Inc I210 Gigabit Network Connection
> 	Kernel driver in use: igb
> 	Kernel modules: igb

Acknowledge. Please send the output of

  lspci -nvs 00:19.0
  lspci -nvs 01:00.0
  lspci -nvs 04:00.0


> livecd ~ # ip l
> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
>     link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
> 2: eno1: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN mode DEFAULT group default qlen 1000
>     link/ether 0c:c4:7a:7d:9f:a0 brd ff:ff:ff:ff:ff:ff
>     altname enp4s0
> 3: eno2: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc pfifo_fast state DOWN mode DEFAULT group default qlen 1000
>     link/ether 0c:c4:7a:7d:9f:a1 brd ff:ff:ff:ff:ff:ff
>     altname enp0s25
> 4: enp1s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
>     link/ether 0c:c4:7a:bb:17:b2 brd ff:ff:ff:ff:ff:ff
> livecd ~ # 

Three NICs, only one Network Interface Card has "UP" and "LOWER_UP".
Please tell more about that NIC.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Reply by alarig** [reply]
Sorry for the late reply. The NICs in no-carrier aren’t plugged so this state is expected, the one that is plugged is `0c:c4:7a:bb:17:b2`. Here is the lscpi output:
```
~# lspci -nvs 01:00.0
01:00.0 0200: 8086:1557 (rev 01)
	Subsystem: 15d9:0659
	Flags: bus master, fast devsel, latency 0, IRQ 16
	Memory at f7e00000 (64-bit, prefetchable) [size=512K]
	I/O ports at e000 [size=32]
	Memory at f7e80000 (64-bit, prefetchable) [size=16K]
	Expansion ROM at f7200000 [disabled] [size=512K]
	Capabilities: [40] Power Management version 3
	Capabilities: [50] MSI: Enable- Count=1/1 Maskable+ 64bit+
	Capabilities: [70] MSI-X: Enable+ Count=64 Masked-
	Capabilities: [a0] Express Endpoint, MSI 00
	Capabilities: [e0] Vital Product Data
	Capabilities: [100] Advanced Error Reporting
	Capabilities: [140] Device Serial Number 0c-c4-7a-ff-ff-bb-17-b2
	Capabilities: [150] Alternative Routing-ID Interpretation (ARI)
	Capabilities: [160] Single Root I/O Virtualization (SR-IOV)
	Kernel driver in use: ixgbe
```

I saw that there are some commits on the driver since the last release, I’ll have to check if that fixes it or not.

**Reply by stappersg** [reply]
```json
{
   "this_is": "some sort of summary",
   "lspci": "01:00.0 Ethernet controller: Intel Corporation 82599 10 Gigabit Network Connection (rev 01)",
   "MAC_address": "0c:c4:7a:bb:17:b2",
   "linux_device_name": "enp1s0",
   "LOWER_UP": true
}   
```
@alarig please include the _iPXE (error) messages_ to your feedback on latest version.
